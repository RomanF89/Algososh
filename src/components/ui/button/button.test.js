import renderer from 'react-test-renderer';
import { render, screen, fireEvent } from '@testing-library/react';

import { Button } from './button';

describe('Кнпока рендерится без ошибок', () => {
  it('Кнопка с текстом рендерится без ошибок', () => {
    const tree = renderer
      .create(<Button text="Добавить в head"/>)
      .toJSON();
      expect(tree).toMatchSnapshot();
  }); 
  it('Кнопка без текста рендерится без ошибок', () => {
    const tree = renderer
      .create(<Button/>)
      .toJSON();
      expect(tree).toMatchSnapshot();
  }); 
  it('Заблокированная кнопка рендерится без ошибок', () => {
    const tree = renderer
      .create(<Button disabled={true}/>)
      .toJSON();
      expect(tree).toMatchSnapshot();
  }); 
  it('Кнопка с индикацией загрузки рендерится без ошибок', () => {
    const tree = renderer
      .create(<Button isLoader={true}/>)
      .toJSON();
      expect(tree).toMatchSnapshot();
  }); 
  it('Корректность вызова коллбэка при клике на кнопку', () => {
    window.alert = jest.fn();
    render(<Button onClick={() => alert('Кнопка работает!')} text='Кнопка коллбэка'/>);
    const button = screen.getByText('Кнопка коллбэка');
    fireEvent.click(button);
    expect(window.alert).toHaveBeenCalledWith('Кнопка работает!');
  }); 
})

