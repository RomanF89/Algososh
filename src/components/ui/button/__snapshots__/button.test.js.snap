// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Кнопка с текстом рендерится без ошибок 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  >
    Добавить в head
  </p>
</button>
`;

exports[`Кнпока рендерится без ошибок Заблокированная кнопка рендерится без ошибок 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  disabled={true}
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Кнпока рендерится без ошибок Кнопка без текста рендерится без ошибок 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Кнпока рендерится без ошибок Кнопка без текстоа рендерится без ошибок 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Кнпока рендерится без ошибок Кнопка с индикацией загрузки рендерится без ошибок 1`] = `
<button
  className="text text_type_button text_color_primary button undefined loader "
  disabled={true}
  type="button"
>
  <img
    alt="Загрузка."
    className="loader_icon"
    src="loader.svg"
  />
</button>
`;

exports[`Кнпока рендерится без ошибок Кнопка с текстом рендерится без ошибок 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  >
    Добавить в head
  </p>
</button>
`;
